src/main/java/com/iexec/sms/secret/MeasuredSecretService.java

```diff
@@ -0,0 +1,113 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.secret;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Metrics;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.PostConstruct;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+/**
+ * Centralisation of all secrets metrics for a type of secret.
+ * <p>
+ * E.g., initial count, number of secrets added since start and currently stored secrets
+ * for `web2Secrets` are metrics that can be retrieved there.
+ * <p>
+ * Stored secrets count is a scheduled operation, so it can reflect a previous state.
+ * That's required to avoid any kind of DDoS.
+ */
+@Getter
+@Slf4j
+public class MeasuredSecretService {
+    private static final String INITIAL_SECRETS_COUNT_POSTFIX = "initial";
+    private static final String ADDED_SECRETS_SINCE_START_COUNT_POSTFIX = "added";
+    private static final String STORED_SECRETS_COUNT_POSTFIX = "stored";
+
+    private final String secretsType;
+    private final String metricsPrefix;
+    private final Supplier<Long> storedSecretsCountGetter;
+    private final ScheduledExecutorService secretsCountExecutorService;
+    private final long storedSecretsCountPeriod;
+
+    private Counter initialSecretsCounter;
+    private AtomicLong storedSecretsCount;
+    private final Counter addedSecretsSinceStartCounter;
+
+    public MeasuredSecretService(String secretsType,
+                                 String metricsPrefix,
+                                 Supplier<Long> storedSecretsCountGetter,
+                                 ScheduledExecutorService secretsCountExecutorService,
+                                 long storedSecretsCountPeriod) {
+        this.secretsType = secretsType;
+        this.metricsPrefix = metricsPrefix;
+        this.storedSecretsCountGetter = storedSecretsCountGetter;
+        this.secretsCountExecutorService = secretsCountExecutorService;
+        this.storedSecretsCountPeriod = storedSecretsCountPeriod;
+
+        this.addedSecretsSinceStartCounter = Metrics.counter(metricsPrefix + ADDED_SECRETS_SINCE_START_COUNT_POSTFIX);
+    }
+
+    @PostConstruct
+    void init() {
+        final long initialSecretsCount = storedSecretsCountGetter.get();
+        this.initialSecretsCounter = Metrics.counter(metricsPrefix + INITIAL_SECRETS_COUNT_POSTFIX);
+        this.initialSecretsCounter.increment(initialSecretsCount);
+
+        this.storedSecretsCount = Metrics.gauge(metricsPrefix + STORED_SECRETS_COUNT_POSTFIX, new AtomicLong(initialSecretsCount));
+
+        secretsCountExecutorService.scheduleAtFixedRate(
+                this::countStoredSecrets,
```

Comment:
It may throw, right. IMHO, the service should return "-1", as it is not an usual value.
Here is a fix: https://github.com/iExecBlockchainComputing/iexec-sms/pull/185/commits/80a5a0cb369a4307fa0421516b66f75e0c729f43